### PR TITLE
fix: geometry changes were not reflected when HDRP pathtracing is on

### PR DESCRIPTION
**Pull request by sindharta**

### CODE REVIEW

File: Runtime/Scripts/BaseMeshSync.cs [hunk 1]
```diff
@@ -2716,7 +2761,14 @@ private protected virtual void OnDisable()
         SceneView.duringSceneGui -= OnSceneViewGUI;
 #endif
     }
-#endregion
+
+    private void Update() {
+#if AT_USE_HDRP
+        UpdatePathTracingState();
+#endif
```
Comment (seandillon92) [inline Runtime/Scripts/BaseMeshSync.cs:2768]: Does the number of servers in the scene affect performance?

File: Runtime/Scripts/BaseMeshSync.cs [hunk 2]
```diff
@@ -1330,9 +1352,32 @@ private EntityRecord UpdateMeshEntity(MeshData data, MeshSyncPlayerConfig config
...
+#if AT_USE_HDRP
 
+        void UpdatePathTracingState() {
+            m_pathTracingExists = false;
+            if (!HDRPUtility.IsRayTracingActive()) 
+                return;
+            
+            SceneComponents<Volume> sceneVolumes = SceneComponents<Volume>.GetInstance();
+            sceneVolumes.Update();
+            m_pathTracingExists = HDRPUtility.IsPathTracingActive(sceneVolumes.GetCachedComponents());
```
Comment (seandillon92) [inline Runtime/Scripts/BaseMeshSync.cs:1367]: What exactly does this code do? seems like its updating scene volumes per frame. How is this related to MeshSync?

**Review by seandillon92 — COMMENTED** [review]
Some questions about why we update the scene volumes

File: Runtime/Scripts/BaseMeshSync.cs (hunk 1 above)
Comment (sindharta) [reply Runtime/Scripts/BaseMeshSync.cs:2768]: No, in theory, because `Object.FindObjectsOfType` is only done once per frame at most

File: Runtime/Scripts/BaseMeshSync.cs (hunk 2 above)
Comment (sindharta) [reply Runtime/Scripts/BaseMeshSync.cs:1367]: `Update()` here means updating cached components, so that we only do `Object.FindObjectsOfType` per frame

Comment (seandillon92) [reply Runtime/Scripts/BaseMeshSync.cs:1367]: Why do we cache and update these components?

Comment (seandillon92) [reply Runtime/Scripts/BaseMeshSync.cs:1367]: I suppose this should be implemented by the user of the package, but we want to allow users to get PathTracing working when using MeshSync without any effort from their side?

Comment (sindharta) [reply Runtime/Scripts/BaseMeshSync.cs:1367]: > Why do we cache and update these components?

Because PathTracing is a property in a `VolumeProfile` assigned to a `Volume` component. 
Meaning that we need to check any `Volume` in the scene if they have PathTracing enabled.

PathTracing should not be confused with RayTracing, which is an HDRP setting.

> I suppose this should be implemented by the user of the package, but we want to allow users to get PathTracing working when using MeshSync without any effort from their side?

That's correct.

**Review by seandillon92 — APPROVED** [review]
